skyxplore-ui-test/src/test/java/com/mentoring/pageobject/EquipmentPage.java

```diff
@@ -19,4 +44,53 @@ public static String getPageUrl() {
     EquipmentPage waitUntilPageLoads() {
         return this;
     }
-}
\ No newline at end of file
+
+    public void removeShipBat01Item() {
+        log.info("Removing BAT-01 item from Ship section by clicking it");
+        waitUntilClickable(shipBat01Item).click();
+    }
+
+    @Step("Moving CEX-01 item from Storage section to Ship empty slot by dragging and dropping it")
+    public void moveStorageCex01ItemToShipEmptySlot() {
+        dragAndDrop(waitUntilVisible(storageCex01Item), waitUntilVisible(shipEmptySlot));
+    }
+
+    public int getShipCex01EmptySlotAttribute() {
+        String wholeAttr, emptySlotAttr;
+        String text = "Extra hely: ";
+
+        wholeAttr = waitUntilVisible(shipCex01Item).getAttribute("title");
+        int pos = wholeAttr.indexOf(text) + text.length();
+        emptySlotAttr = wholeAttr.substring(pos);
+        return Integer.parseInt(emptySlotAttr);
+    }
```

Comment:
1. The name of the method is not descriptive enough. It's hard to find out what it's doing. getNumberOfFreeSlotsFroCex01ItemDescription is better for example.
2. Step annotation is missing.
3. Use regex instead of substring position. You can find example here: https://stackoverflow.com/questions/14344275/regex-to-match-number-of-1-3-digits-after-a-known-string-in-ruby

Reply:
Fixed. Thank you for your helping related to the correct regex! :-)